### PR TITLE
Fix Doxygen/docs links and expand documentation index page

### PR DESCRIPTION
**Pull request by enirolf**

### CODE REVIEW

File: for_developers/docs/index.md [hunk 1]
```diff
@@ -8,5 +8,4 @@ sidebar:
 ---
 
 The ROOT team has adopted [Doxygen](https://www.doxygen.nl){:target="_blank"} for generating
-the [Reference Guide](https://root.cern/doc/master/){:target="_blank"}. In this section, you will find a number of conventions
-and tips to be used in source code to generate proper documentation.
+the [Reference Guide](https://root.cern/doc/master/){:target="_blank"}. In this section, you will find [our documentation guidelines](./guidelines.md), how to [run Doxygen locally](./local_doxygen.md), how to add [documentation for ROOT's pythonizations](./pythonizations.md) and a number of [useful tips formatting tips](./useful_tips.md).
```
Comment (martamaja10) [inline for_developers/docs/index.md:11]: Those links don't work for me when I try it on: https://root.cern/1052/for_developers/docs/

**Review by martamaja10 — COMMENTED** [review]
Thanks a lot - a few links don't work for me though.

File: for_developers/docs/index.md (hunk 1 above)
Comment (martamaja10) [reply for_developers/docs/index.md:11]: [our documentation guidelines], [run Doxygen locally], [documentation for ROOT’s pythonizations] and [useful tips formatting tips]

Comment (enirolf) [reply for_developers/docs/index.md]: Whoops thanks for the catch -- should be fixed now!

**Review by martamaja10 — APPROVED** [review]
Great, all looks good now!

**Pull request merged**